client/src/app/gateways/repositories/structure-levels/structure-level-repository.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core';
+
+import { Identifiable } from '../../../domain/interfaces';
+import { StructureLevel } from '../../../domain/models/structure-levels/structure-level';
+import { ViewStructureLevel } from '../../../site/pages/meetings/pages/participants/pages/structure-levels/view-models/view-structure-level';
+import { BaseMeetingRelatedRepository } from '../base-meeting-related-repository';
+import { RepositoryMeetingServiceCollectorService } from '../repository-meeting-service-collector.service';
+import { StructureLevelAction } from './structure-level.action';
+
+@Injectable({
+    providedIn: `root`
+})
+export class StructureLevelRepositoryService extends BaseMeetingRelatedRepository<ViewStructureLevel, StructureLevel> {
+    public constructor(repositoryServiceCollector: RepositoryMeetingServiceCollectorService) {
+        super(repositoryServiceCollector, StructureLevel);
+    }
+
+    public getVerboseName = (plural?: boolean): string => (plural ? `StructureLevels` : `StructureLevel`);
+    public getTitle = (viewModel: ViewStructureLevel): string => viewModel.name;
+
+    public async create(...structureLevels: Partial<StructureLevel>[]): Promise<Identifiable[]> {
+        const payload = structureLevels.map(structureLevel => ({
+            name: structureLevel.name,
+            color: structureLevel.color,
+            allow_additional_time: structureLevel.allow_additional_time,
+            meeting_id: this.activeMeetingId
+        }));
+        return this.sendBulkActionToBackend(StructureLevelAction.CREATE, payload);
```

Comment:
sendBulkActionToBackend is deprecated, use `createAction` or something like that instead.
Return the Action object, not a Promise

Reply:
Okay, use `createAction` here and for `update` and `delete`, too.